python/cuml/fil/fil.pyx

```diff
@@ -215,7 +240,11 @@ cdef class TreeliteModel():
         """
         assert self.handle != NULL
         filename_bytes = filename.encode("UTF-8")
-        TreeliteSerializeModelToFile(self.handle, filename_bytes)
+        cdef int res = TreeliteSerializeModelToFile(self.handle, filename_bytes)
+        cdef const char* err
+        if res < 0:
+            err = TreeliteGetLastError()
+            raise RuntimeError("Failed to serialize Treelite model (%s)" % (err))
```

Comment:
The formatting here might look a little odd as it will represent it as `bytes`

```python
RuntimeError: Failed to serialize Treelite model (b'My Error')
```

Though if convert the error message to a string first, it should look better

```python
RuntimeError: Failed to serialize Treelite model ('My Error')
```

Hence this suggestion and the suggestions above

```suggestion
            raise RuntimeError("Failed to serialize Treelite model (%s)" % (err_msg))
```